services/core/shared/core/delegates.js

```diff
@@ -14,13 +14,23 @@
  *
  */
 const { Logger } = require('lisk-service-framework');
+const BluebirdPromise = require('bluebird');
 
 const config = require('../../config');
 const pouchdb = require('../pouchdb');
 const coreApi = require('./compat');
+const { getBlocks } = require('./blocks');
 
 const logger = Logger();
 
+let nextForgers = [];
+
+const delegateComparator = (a, b) => {
+	const diff = b.delegateWeight - a.delegateWeight;
+	if (diff !== 0) return diff;
+	return Buffer.from(a.account.address).compare(Buffer.from(b.account.address));
```

Comment:
Why is this a `Buffer` comparison? Wouldn't a regular `Number` be enough?

Reply:
Lisk SDK uses similar logic to rank the delegates. The ranking logic sorts the delegates based on their `delegateWeight`. And, in the case of a tie, the ordering is done by buffer based sorting of their addresses.

Refer: Lisk SDK implementation ([here](https://github.com/LiskHQ/lisk-sdk/blob/2bf7defd9a94ba2bc803e48c98c3a3e3fcf90854/framework/src/modules/dpos/delegates.ts#L34))